tests/gha-run-k8s-common.sh

```diff
@@ -94,6 +94,25 @@ function install_kubectl() {
     sudo az aks install-cli
 }
 
+# Install the kustomize tool in /usr/local/bin if it doesn't exist on
+# the system yet.
+#
+function install_kustomize() {
+	local version
+	local arch
+
+	if command -v kustomize >/dev/null; then
+		return
+	fi
+
+	ensure_yq
+	version=$(get_from_kata_deps "externals.kustomize.version")
+	arch=$(arch_to_golang)
+
+	curl -L "https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize/${version}/kustomize_${version}_linux_${arch}.tar.gz" | \
+		sudo tar xvzf - -C /usr/local/bin
```

Comment:
I think we should be paranoid here:

- Download the tar file to a `mktemp -d` directory.
- Download the checksum file (for example, https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%2Fv5.3.0/checksums.txt).
- Check the tar file against the checksum.
- Assuming the checksum is correct, unpack and then delete the temp directory.

Reply:
what about also implementing a trap to ensure that the temp directory is deleted
```
TEMP_DIR="$( mktemp -d )"
 trap 'rm -rf ${TEMP_DIR}' EXIT
```